discovery-infra/test_infra/helper_classes/kube_helpers.py

```diff
@@ -0,0 +1,376 @@
+
+    @property
+    def api_client(self) -> ApiClient:
+        return self._shared_api_client[0]
+
+    @classmethod
+    def _set_api_client_if_needed(cls) -> None:
+        # reuse same client for all subclasses.
+        if cls._shared_api_client:
+            return
+
+        conf = Configuration()
+        load_kube_config(client_configuration=conf)
+        cls._shared_api_client.append(ApiClient(conf))
+
+
+class Secret(_BaseKubeAPIResource):
+
+    def __init__(self, name: str, namespace: str = env_variables['namespace']):
+        _BaseKubeAPIResource.__init__(self, name, namespace)
+
+        self.v1_api = CoreV1Api(self.api_client)
+
+    def create(self, pull_secret: str = env_variables['pull_secret']) -> None:
+        self.v1_api.create_namespaced_secret(
+            body={
+                'apiVersion': 'v1',
+                'kind': 'Secret',
+                'metadata': self.ref,
+                'stringData': {'pullSecret': pull_secret}
+            },
+            namespace=self.ref['namespace']
+        )
+
+        logger.info('created secret %s', self.ref)
+
+    def delete(self) -> None:
+        self.v1_api.delete_namespaced_secret(
+            name=self.ref['name'],
+            namespace=self.ref['namespace']
+        )
+
+        logger.info('deleted secret %s', self.ref)
+
+
+DEFAULT_API_VIP = env_variables.get('api_vip', '')
+DEFAULT_API_VIP_DNS_NAME = env_variables.get('api_vip_dns_name', '')
+DEFAULT_INGRESS_VIP = env_variables.get('ingress_vip', '')
+
+
+class Platform(dict):
+
+    def __init__(
+            self,
+            api_vip: str = DEFAULT_API_VIP,
+            api_vip_dns_name: str = DEFAULT_API_VIP_DNS_NAME,
+            ingress_vip: str = DEFAULT_INGRESS_VIP,
+            vip_dhcp_allocation: bool = env_variables['vip_dhcp_allocation']
+    ):
+        dict.__init__(
+            self,
+            agentBareMetal={
+                'apiVIP': api_vip,
+                'ingressVIP': ingress_vip,
+                'VIPDHCPAllocation': 'Enabled' if vip_dhcp_allocation else ''
+            }
+        )
+
+        if _does_string_contain_value(api_vip_dns_name):
+            self['agentBareMetal']['apiVIPDNSName'] = api_vip_dns_name,
+
+
+DEFAULT_MACHINE_CIDR = env_variables.get('machine_cidr', '')
+DEFAULT_CLUSTER_CIDR = env_variables.get('cluster_cidr', '172.30.0.0/16')
+DEFAULT_SERVICE_CIDR = env_variables.get('service_cidr', '10.128.0.0/14')
+
+
+class InstallStrategy(dict):
+
+    def __init__(
+            self,
+            control_plane_agents: int = 1,
+            worker_agents: int = 0,
+            label_selector: Optional[Dict[str, str]] = None,
+            host_prefix: int = env_variables['host_prefix'],
+            machine_cidr: str = DEFAULT_MACHINE_CIDR,
+            cluster_cidr: str = DEFAULT_CLUSTER_CIDR,
+            service_cidr: str = DEFAULT_SERVICE_CIDR,
+            ssh_public_key: str = env_variables['ssh_public_key'],
+    ):
+        dict.__init__(
```

Comment:
`super().__init__(....)`

Reply:
why?

Reply:
I mean why do you insist on it?

Reply:
IMO it seems more clean and Pythonic. But after searching a little bit on the web, it seems that there are people that might disagree https://fuhm.net/super-harmful/
You can keep. I will keep searching about it :)

Reply:
thanks :)